src/arb_mat/spd_is_lll_reduced.c

```diff
@@ -0,0 +1,57 @@
+/*
+    Copyright (C) 2023 Jean Kieffer
+
+    This file is part of Arb.
+
+    Arb is free software: you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.  See <http://www.gnu.org/licenses/>.
+*/
+
+#include "fmpz_mat.h"
+#include "fmpz_lll.h"
+#include "arb_mat.h"
+
+int arb_mat_spd_is_lll_reduced(const arb_mat_t A, slong tol_exp, slong prec)
+{
+    slong g = arb_mat_nrows(A);
+    arb_mat_t B;
+    fmpz_mat_t N;
+    arb_t c;
+    int res = 1;
+    slong j, k;
+
+    arb_mat_init(B, g, g);
+    fmpz_mat_init(N, g, g);
+    arb_init(c);
+
+    /* Set B, check error bounds on coefficients */
+    for (j = 0; (j < g) && res; j++)
+    {
+        for (k = 0; (k < g) && res; k++)
+        {
+            if (mag_cmp_2exp_si(arb_radref(arb_mat_entry(A, j, k)), tol_exp - 4) > 0)
+            {
+                res = 0;
```

Comment:
Apparently missing test coverage: would it be difficult to produce a test case that reaches this branch?

Reply:
This should be fixed in https://github.com/flintlib/flint/pull/1524/commits/4a78178e4ecdb09628b169412c182ce66c4a1b5b.
Maybe a better definition of `arb_mat_spd_is_lll_reduced` would be to more or less copy-paste code from `fmpz_lll_is_reduced_mpfr`. I could also add an LLL context as an argument instead of choosing default values. The current versions are sufficient for the `acb_theta` module though.